x-pack/filebeat/input/o365audit/auth/cert.go

```diff
@@ -57,9 +48,9 @@ func loadConfigCerts(cfg tlscommon.CertificateConfig) (cert *x509.Certificate, k
 	if tlsCert.PrivateKey == nil {
 		return nil, nil, fmt.Errorf("failed loading private key from '%s'", cfg.Key)
 	}
-	key, ok := tlsCert.PrivateKey.(*rsa.PrivateKey)
+	_, ok := tlsCert.PrivateKey.(*rsa.PrivateKey)
 	if !ok {
 		return nil, nil, fmt.Errorf("private key at '%s' is not an RSA private key", cfg.Key)
 	}
-	return cert, key, nil
+	return cert, tlsCert.PrivateKey, nil
```

Comment:
Suggest reverting the change to loadConfigCerts. This gives us immediate documentation of the type of `crypto.PrivateKey` we have. Maxim that applies here: "Accept interfaces, return concrete types."

Reply:
Yes, that was unnecessary. I reverted it.